client/src/components/Categories/SingleCategory.js

```diff
@@ -0,0 +1,40 @@
+import React, { Component } from 'react';
+import apiClient from '../../helpers/apiClient'
+
+class SingleCategory extends Component {
+
+    constructor() {
+
+        super();
+        this.state = {
+            articles: []
+        }
+    }
+    componentDidMount() {
+        apiClient.getArticles()
```

Comment:
This works but it is inefficient. Right now, you're getting all articles and then filtering them on the client-side. This is expensive operation and unnecessary.

Looking at changing `/articles` API - so that you can call it saying `/articles?categoryId=_id`, then you can get the category Id on the server from `request.params` and query the database directly by `{category: categoryID}`

Does that make sense?

Reply:
is the same have done with singleArticle?

Reply:
Do I need to change one api/articles route in the server side?

Reply:
I lost actually

Reply:
you can /api/articles so that it reads req.params (which will contain {cateogry: _id} if you call /api/articles?category=_id  .. then it can pass that object to Mongo which will filter by category.

Reply:
maybe it is `req.queryString` actually not req.params - test and see

Reply:
sorry for replay late just because I am on the college I am doing coding when I am free, so alright I will try to do what you mentioned.

Reply:
That is did not work but I will carry on to understand how is it working.